airflow/models/dagrun.py

```diff
@@ -614,6 +620,7 @@ def _check_last_n_dagruns_failed(self, dag_id, max_consecutive_failed_dag_runs,
     def get_task_instances(
         self,
         state: Iterable[TaskInstanceState | None] | None = None,
+        dag_run_option: Literal["lazy", "joined"] = "joined",
```

Comment:
Maybe something like `load_run_option` instead? This sounds like an option to DAG run instead of db-loading.

I kind of feel we should just avoid trying to abstract this sort of things in the first place too. Using SQL (or ORM) calls directly is more often the correct abstraction.

Reply:
I agree. Too much dry gets us into trouble like this

Reply:
so would you say, @uranusjr , in this case, we should just inline the querying for this use case?  E.g. duplicate [only the needed bits of] this to the DepContext class?
one complicating factor is that this ultimately needs to be rpc-compatible.   so it needs to be enclosed in a function somehow.  but, it doesn't have to be a function shared by anything else.  so it could be `DepContext._get_task_finished_tis`  for example.

Reply:
on your other point, i chose `dag_run_option` because there's another table in there (and who knows there could be others in future)

Reply:
@uranusjr gentle nudge

Reply:
Let’s either inline the query and not touch this function.